src/Version/CMakeLists.txt

```diff
@@ -53,4 +53,24 @@ set_source_files_properties("${version_inc}"
              HEADER_FILE_ONLY TRUE
   )
 
-add_custom_target(OMVersion ALL DEPENDS ${version_inc})
+if (NOT DEFINED ONNX_MLIR_VERSION)
+  set(ONNX_MLIR_VERSION "0.3.0")
+endif()
+list(APPEND DEFINITIONS "ONNX_MLIR_VERSION=\"${ONNX_MLIR_VERSION}\"")
+if (NOT DEFINED ONNX_VERSION)
+  set(ONNX_VERSION "1.11.0")
```

Comment:
Can't the default be set based on what's in the submodule instead of having to be explicitly specified here? It seems like this would be prone to error.

Reply:
In `build/third_party/onnx/ONNXConfig.cmake`, `set(ONNX_VERSION "1.11.0")`.
I can probably use that, I am not familiar with cmake, so I need to find out how to do that.

Reply:
This should do what you need.
```
execute_process(
  COMMAND git tag --points-at HEAD
  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/onnx
  OUTPUT_VARIABLE ONNX_VERSION
  OUTPUT_STRIP_TRAILING_WHITESPACE
  )
if ("${ONNX_VERSION}" STREQUAL "")
  message(WARNING "third_party/onnx version not detected, you may be on the wrong commit hash")
else()
  # Strip leading "v"
  STRING(REGEX REPLACE "^v" "" ONNX_VERSION ${ONNX_VERSION})
  message(STATUS "Onnx version             : ${ONNX_VERSION}")
endif()
```

Reply:
Thanks @gongsu832 that save me a lot of time.

Reply:
On MacOS:
```
CMake Warning at src/Version/CMakeLists.txt:75 (message):
  third_party/onnx version not detected, you may be on the wrong commit hash
```
Will investigate why.

Reply:
Changed to `cat VERSION_NUMBER`, and that works on all platforms.